src/main/scala/org/neo4j/spark/Neo4jSparkSession.scala

```diff
@@ -0,0 +1,184 @@
+     * @param parameters
+     * @param schema
+     * @return
+     * @example     import org.apache.spark.sql.types._
+     *              import org.neo4j.spark.Neo4jSparkSession
+     *              Neo4jSparkSession.builder.getOrCreate().DF.
+     *                query("MATCH (n) WHERE id(n) < {maxId} return n.name as name",Seq("maxId" -> 100000),"name" -> "string").count
+     */
+    def query(cquery: String, parameters: Seq[(String, Any)], schema: (String, String)) = {
+      Neo4jDataFrame(sqlContext, cquery, parameters, schema)
+    }
+
+  }
+
+  /**
+   * Contains all Neo4j Spark Graph operations
+   */
+  object Graph{
+    def loadGraph[VD:ClassTag,ED:ClassTag](nodeStmt: (String,Seq[(String, AnyRef)]), relStmt: (String,Seq[(String, AnyRef)])) :Graph[VD,ED] = {
+      Neo4jGraph.loadGraph(sc, nodeStmt, relStmt)
+    }
+
+    def loadGraph[VD:ClassTag,ED:ClassTag](nodeStmt: String, relStmt: String) :Graph[VD,ED] = {
+      Neo4jGraph.loadGraph(sc,nodeStmt, relStmt)
+    }
+
+
+    def loadGraph(label1: String, relTypes: Seq[String], label2: String) : Graph[Any,Int] = {
+      Neo4jGraph.loadGraph(sc,label1, relTypes, label2)
+    }
+
+    def loadGraphFromRels[VD:ClassTag,ED:ClassTag](statement: String, parameters: Seq[(String, AnyRef)], defaultValue : VD = Nil) :Graph[VD,ED] = {
+      Neo4jGraph.loadGraphFromRels(sc, statement, parameters, defaultValue)
+    }
+
+    def loadGraphFromNodePairs[VD:ClassTag](statement: String, parameters: Seq[(String, AnyRef)] = Seq.empty, defaultValue : VD = Nil) :Graph[VD, Int] = {
+      Neo4jGraph.loadGraphFromNodePairs(sc, statement, parameters, defaultValue)
+    }
+
+    def saveGraph[VD:ClassTag,ED:ClassTag](graph: Graph[VD,ED], nodeProp : String = null, relProp: String = null) : (Long,Long) = {
+      Neo4jGraph.saveGraph(sc, graph, nodeProp, relProp)
+    }
+
+  }
+}
+
+/**
+ * @author Mageswaran Dhandapani
+ * @since 09.07.16
+ *
+ * Case 1: Retrieves spark-shell context
+ * Case 2: Retrieves application context
+ * Case 3: New context is created with user given bolt password
+ *
+ * @example val ns = Neo4jSparkSession.builder.sparkContext(sc).boltPassword("neo4j").getOrCreate()
+ */
+object Neo4jSparkSession {
+
+  class Builder {
+
+    val session: Option[Neo4jSparkSession] = None
+
+    private[this] var userSpecifiedContext: Option[SparkContext] = None
+    private[this] var userSpecifiedSqlContext: Option[SQLContext] = None
+    private[this] var userSpecifiedSparkConf: Option[SparkConf] = None
+    private[this] var userSpecifiedBoltPassword: Option[String] = None
+    //private[this] var userSpecifiedSession: Option[SparkSession] = None //TODO SparkSession
+
+    val sparkContext = userSpecifiedContext.getOrElse {
+      val sparkConf =userSpecifiedSparkConf.getOrElse {
+        //User password is used when started inside an application
+        new SparkConf().set("spark.neo4j.bolt.password", userSpecifiedBoltPassword.getOrElse(""))
```

Comment:
After testing, adding "throw" will enforce user from spark shell to use the boltPassword() call eventhough the config comes from the default spark config. What you think?